spyder/widgets/panels/manager.py

```diff
@@ -0,0 +1,308 @@
+            Panel.Position.LEFT: 'left',
+            Panel.Position.RIGHT: 'right',
+            Panel.Position.TOP: 'top'
+        }
+        _logger().log(5, 'adding panel %s at %r', panel.name,
+                      pos_to_string[position])
+        panel.order_in_zone = len(self._panels[position])
+        self._panels[position][panel.name] = panel
+        panel.position = position
+        panel.on_install(self.editor)
+        _logger().log(5, 'panel %s installed', panel.name)
+        return panel
+
+    def remove(self, name_or_klass):
+        """
+        Removes the specified panel.
+
+        :param name_or_klass: Name or class of the panel to remove.
+        :return: The removed panel
+        """
+        _logger().log(5, 'removing panel %r', name_or_klass)
+        panel = self.get(name_or_klass)
+        panel.on_uninstall()
+        panel.hide()
+        panel.setParent(None)
+        return self._panels[panel.position].pop(panel.name, None)
+
+    def clear(self):
+        """
+        Removes all panel from the CodeEditor.
+        """
+        for i in range(4):
+            while len(self._panels[i]):
+                key = sorted(list(self._panels[i].keys()))[0]
+                panel = self.remove(key)
+                panel.setParent(None)
+                panel.deleteLater()
+
+    def get(self, name_or_klass):
+        """
+        Gets a specific panel instance.
+
+        :param name_or_klass: Name or class of the panel to retrieve.
+        :return: The specified panel instance.
+        """
+        if not isinstance(name_or_klass, str):
+            name_or_klass = name_or_klass.__name__
+        for zone in range(4):
+            try:
+                panel = self._panels[zone][name_or_klass]
+            except KeyError:
+                pass
+            else:
+                return panel
+        raise KeyError(name_or_klass)
+
+    def keys(self):
+        """
+        Returns the list of installed panel names.
+        """
+        return self._modes.keys()
+
+    def values(self):
+        """
+        Returns the list of installed panels.
+        """
+        return self._modes.values()
+
+    def __iter__(self):
+        lst = []
+        for zone, zone_dict in self._panels.items():
+            for name, panel in zone_dict.items():
+                lst.append(panel)
+        return iter(lst)
+
+    def __len__(self):
+        lst = []
+        for zone, zone_dict in self._panels.items():
+            for name, panel in zone_dict.items():
+                lst.append(panel)
+        return len(lst)
+
+    def panels_for_zone(self, zone):
+        """
+        Gets the list of panels attached to the specified zone.
+
+        :param zone: Panel position.
+
+        :return: List of panels instances.
+        """
+        return list(self._panels[zone].values())
+
+    def refresh(self):
+        """ Refreshes the editor panels (resize and update margins) """
```

Comment:
Missing dot at the end, remove blanks at the beginning and end of this docstring.